source/extensions/omni.isaac.lab/omni/isaac/lab/managers/observation_manager.py

```diff
@@ -14,6 +14,7 @@
 from typing import TYPE_CHECKING
 
 from omni.isaac.lab.utils import modifiers
+from omni.isaac.lab.utils.buffers import CircularBuffer
 
 from .manager_base import ManagerBase, ManagerTermBase
 from .manager_term_cfg import ObservationGroupCfg, ObservationTermCfg
```

Comment:
I cannot find how to provide a suggestion to lines that are not part of the changelist, so here we go.

In the method `def __str__(self) -> str:`, 

You may need to re-compute the observation dimensions depending on whether flattening is enabled or not. 
I see two deficiencies which can cause confusions:
1. Currently, the printed summary does not handle history length while computing the observation dimension. 
2. The `self._group_obs_dim` is wrong and does not correspond to the actual observation dimension if history is used. This is quite critical to be fixed IMO.
3. The shape of the observation group printed in the summary is wrong.

Hope this helps!

Reply:
Thanks for the catch I will address this.